### PR TITLE
Tdr 2488 move GitHub terraform from backend repo

### PR DESCRIPTION
**Pull request by vimleshtna**

### CODE REVIEW

File: modules/state-lock/variables.tf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+variable "common_tags" {
```
Comment (TomJKing) [inline modules/state-lock/variables.tf:1]: This needs to go in the tdr-terraform-backend repo

File: modules/state/output.tf [hunk 2]
```diff
@@ -0,0 +1,16 @@
+output "terraform_state_bucket_arn" {
```
Comment (TomJKing) [inline modules/state/output.tf:1]: This needs to go in the tdr-terraform-backend repo

File: modules/state/variables.tf [hunk 3]
```diff
@@ -0,0 +1,4 @@
+variable "common_tags" {
```
Comment (TomJKing) [inline modules/state/variables.tf:1]: This needs to go in the tdr-terraform-backend repo

File: root.tf [hunk 4]
```diff
@@ -0,0 +1,51 @@
+locals {
+  backend_state_lock   = "tdr-bootstrap-terraform-state-lock"
```
Comment (TomJKing) [inline root.tf:2]: The state bucket and state lock will not be the backend. It this repo will have its own state lock and state bucket.

**Review by TomJKing — COMMENTED** [review]
Not sure where you have got to with this. But have left some comments.

The state s3 bucket and state lock dynamodb table need to be created in the tdr-terraform-backend repo. Not in this repo. The terraform in this repo can't be run unless they exist, so they have to be created in the backend first. Otherwise there is a circular reference going on.

I've created a PR with the state and state lock changes here: https://github.com/nationalarchives/tdr-terraform-backend/pull/248

File: root.tf (hunk 4 above)
Comment (vimleshtna) [reply root.tf]: Yeah, good point. I have replaced it with our own state and state-lock

**Comment by vimleshtna** [comment]
> Not sure where you have got to with this. But have left some comments.
> 
> The state s3 bucket and state lock dynamodb table need to be created in the tdr-terraform-backend repo. Not in this repo. The terraform in this repo can't be run unless they exist, so they have to be created in the backend first. Otherwise there is a circular reference going on.
> 
> I've created a PR with the state and state lock changes here: [nationalarchives/tdr-terraform-backend#248](https://github.com/nationalarchives/tdr-terraform-backend/pull/248)

Right, thanks for adding them into the `tdr-terraform-backend` repo. I have removed them from this repo.

File: root_github_repository.tf [hunk 5]
```diff
@@ -0,0 +1,539 @@
...
+      "TDR_${upper(environment)}_TERRAFORM_ROLE"        = module.configuration.terraform_config[environment]["terraform_role"]
+      "TDR_${upper(environment)}_CUSTODIAN_ROLE"        = module.configuration.terraform_config[environment]["custodian_role"]
+      "TDR_${upper(environment)}_STATE_BUCKET"          = module.configuration.terraform_config[environment]["state_bucket"]
+      "TDR_${upper(environment)}_DYNAMO_TABLE"          = module.configuration.terraform_config[environment]["dynamo_table"]
+      "TDR_${upper(environment)}_TERRAFORM_EXTERNAL_ID" = module.configuration.terraform_config[environment]["terraform_external_id"]
+    }
+  }
+}
+
+#module "github_transfer_frontend_repository" {
```
Comment (TomJKing) [inline root_github_repository.tf:19]: Could you delete the commented out code. It just makes it easier to follow as you add more repositories to this repo.

File: versions.tf [hunk 6]
```diff
@@ -0,0 +1,13 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "4.50.0"
```
Comment (TomJKing) [inline versions.tf:5]: Should upgrade this to the latest version

File: versions.tf [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    }
+    github = {
+      source  = "integrations/github"
+      version = "~> 5.12"
+    }
+  }
+  required_version = ">= 1.3.2"
```
Comment (TomJKing) [inline versions.tf:12]: Should upgrade this to the latest Terraform version. You will need to install that version on your local machine as well. I put some notes about running different versions of Terraform on the same machine on Slack.

**Review by TomJKing — COMMENTED** [review]
Looks good. Thanks. Just some minor comments.

File: root_github_repository.tf (hunk 5 above)
Comment (vimleshtna) [reply root_github_repository.tf]: yeah, right, deleted them, thanks

File: versions.tf (hunk 7 above)
Comment (vimleshtna) [reply versions.tf]: Upgraded it, I'll add the instructions on the readme file in the next PR

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**